pkg/provisioner/kafka/client.go

```diff
@@ -0,0 +1,62 @@
+package client
+
+import (
+	"github.com/Shopify/sarama"
+	"io"
+	"log"
+)
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 . KafkaClient
+type KafkaClient interface {
+	TopicExists(topicName string) (bool, *KafkaError)
+	CreateTopic(topicName string) error
+	Close() error
+}
+
+type kafkaClient struct {
+	Admin   sarama.ClusterAdmin
+}
+
+func NewKafkaClient(brokerAddress string, logWriter io.Writer) (KafkaClient, error) {
+	sarama.Logger = log.New(logWriter, "[Sarama] ", log.LstdFlags)
```

Comment:
This is reassigning a global var multiple times and possibly concurrently.
Better left globally in main()

Reply:
It's reassigning to the same value, though, isn't it?

Reply:
(note that the global vars have been removed)

Reply:
It's a global var defined by sarama itself.
Need to move the assignment back to main()

Reply:
oh, sorry, I thought you were referring to the broker address!
gotcha, moving it back to main